source/Calamari/Scripts/Octopus.Features.IISWebSite_BeforePostDeploy.ps1

```diff
@@ -45,29 +45,54 @@ try {
     }
 }
 
+function Recover-Semaphore {
+	param(
+	[parameter(Mandatory = $true)][string] $SemaphoreId
+	)
+
+	Try	{
+		Write-Verbose "Possible case of semaphore $SemaphoreId having not been released by a terminated/failed process, attempting to take control"
+		$suspectedAbandonedSemaphore = [System.Threading.Semaphore]::OpenExisting($SemaphoreId);
+		$result = $suspectedAbandonedSemaphore.Release()
+		Write-Verbose "Forcing semaphore release, the result was (it should be zero): $result"
+	}
+	Catch [System.UnauthorizedAccessException] {
+		Write-Verbose "Sempahore timeout cleanup failed or wasn't needed"
+	}
+}
+
 function Wait-OnSemaphore {
 	param(
 	[parameter(Mandatory = $true)][string] $SemaphoreId
 	)
 
-	Try
-	{
+	Try	{
+		$attemptCount = 0
+		$maxFailures = 25 #25*5sec sleep = ~2 min
 		$SemaphoreInstance = New-Object System.Threading.Semaphore -ArgumentList 1, 1, $SemaphoreId
 
 		while (-not $SemaphoreInstance.WaitOne(100))
 		{
-			Write-Verbose "Cannot start this IIS website related task yet. There is already another task running that cannot be run in conjunction with any other task. Please wait..."
+			$attemptCount++;
+			Write-Verbose "Cannot start this IIS website related task yet. There is already another task running that cannot be run in conjunction with any other task. Attempt $attemptCount. Please wait..."
```

Comment:
@NickJosevski  Do we think it would be nice to have attempt x out of total attempts ?